Library/Homebrew/os/mac/sdk.rb

```diff
@@ -64,43 +94,31 @@ def sdk_prefix
         ""
       end
 
-      def sdk_paths
-        @sdk_paths ||= begin
-          # Bail out if there is no SDK prefix at all
-          if File.directory? sdk_prefix
-            paths = {}
+      def latest_sdk
+        all_sdks.max_by(&:version)
+      end
 
-            Dir[File.join(sdk_prefix, "MacOSX*.sdk")].each do |sdk_path|
-              version = sdk_path[/MacOSX(\d+\.\d+)u?\.sdk$/, 1]
-              paths[OS::Mac::Version.new(version)] = sdk_path if version.present?
-            end
+      def read_sdk_version(sdk_path)
+        sdk_settings = sdk_path/"SDKSettings.json"
+        sdk_settings_string = sdk_settings.read if sdk_settings.exist?
 
-            # Use unversioned SDK path on Big Sur to avoid issues such as:
-            # https://github.com/Homebrew/homebrew-core/issues/67075
-            # This creates an entry in `paths` whose key is the OS major version
-            sdk_path = Pathname.new("#{sdk_prefix}/MacOSX.sdk")
-            sdk_settings = sdk_path/"SDKSettings.json"
-            if sdk_settings.exist? &&
-               (sdk_settings_string = sdk_settings.read.presence) &&
-               (sdk_settings_json = JSON.parse(sdk_settings_string).presence) &&
-               (version_string = sdk_settings_json.fetch("Version", nil).presence) &&
-               (version = version_string[/(\d+)\./, 1].presence)
-              paths[OS::Mac::Version.new(version)] = sdk_path
-            end
-
-            paths
-          else
-            {}
-          end
+        # Pre-10.14 SDKs
+        sdk_settings = sdk_path/"SDKSettings.plist"
+        if sdk_settings_string.blank? && sdk_settings.exist?
+          result = system_command("plutil", args: ["-convert", "json", "-o", "-", sdk_settings])
+          sdk_settings_string = result.stdout if result.success?
         end
-      end
 
-      # NOTE: This returns a versioned SDK path, even on Big Sur
-      def latest_sdk
-        return if sdk_paths.empty?
+        return if sdk_settings_string.blank?
+        return unless (sdk_settings_json = JSON.parse(sdk_settings_string).presence)
+        return unless (version_string = sdk_settings_json.fetch("Version", nil).presence)
+        return unless (version = Version.new(version_string).major_minor.presence)
```

Comment:
I think if you drop `.major_minor` here so that you only have
```ruby
version = Version.new(version_string).presence
```
you can change `if version.major >= 11` below to `if version >= :big_sur`

Reply:
`major_minor` is intentional so the 10.15.6 SDK is treated the same as 10.15.

Dropping that won't make a comparison with `:big_sur` work - I would have to convert it to a `OS::Mac::Version` object which I could probably do.

Reply:
Yes, sorry, in my head I had already written `OS::Mac::Version`.

Reply:
Yeah. And unfortunately just feeding `10.15.6` into it will fail the comparison `OS::Mac::Version.new("10.15.6") == OS::Mac::Version.new("10.15")` so I definitely need to do regular `Version` w/ `major_minor` first with a later conversion to `OS::Mac::Version`.

But I eventually convert to `OS::Mac::Version` anyway so it's just a case of moving things around.